core/src/components/header/header.ios.scss

```
@@ -130,7 +130,9 @@
  * 2. This will only apply when that content has a collapse header (ion-header[collapse="condense"])
  *
  * We use opacity: 0 to avoid a layout shift.
+ * We target both the attribute and the class in the event that the attribute
+ * is not reflected on the host in some frameworks.
  */
-ion-header:not(.header-collapse-main):has(~ ion-content ion-header[collapse="condense"]) {
+ion-header:not(.header-collapse-main):has(~ ion-content ion-header[collapse="condense"], ~ ion-content ion-header.header-collapse-condense) {
```

One nit pick - can we move this to its own line? I thought that stylelint had a rule for this but I guess not. 

```suggestion
ion-header:not(.header-collapse-main):has(~ ion-content ion-header[collapse="condense"], 
~ ion-content ion-header.header-collapse-condense) {
```